// @dfns/sdk/codegen/datamodel/PolicyManagement/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyObjectFilterKind = exports.PolicyRuleStatus = exports.PolicyControlStatus = exports.PolicyRuleKind = exports.PolicyControlKind = exports.PolicyStatus = exports.PolicyActivityKind = void 0;
// FIXME: Missing documentation for PolicyActivityKind
var PolicyActivityKind;
(function (PolicyActivityKind) {
    // FIXME: Missing documentation for CreatingSignature
    PolicyActivityKind["CreatingSignature"] = "CreatingSignature";
    // FIXME: Missing documentation for TransactionInitiation
    PolicyActivityKind["TransactionInitiation"] = "TransactionInitiation";
    // FIXME: Missing documentation for PaymentInitiation
    PolicyActivityKind["PaymentInitiation"] = "PaymentInitiation";
    // FIXME: Missing documentation for WalletTransferAsset
    PolicyActivityKind["WalletTransferAsset"] = "WalletTransferAsset";
    // FIXME: Missing documentation for WalletBroadcastTransaction
    PolicyActivityKind["WalletBroadcastTransaction"] = "WalletBroadcastTransaction";
    // FIXME: Missing documentation for WalletGenerateSignature
    PolicyActivityKind["WalletGenerateSignature"] = "WalletGenerateSignature";
})(PolicyActivityKind = exports.PolicyActivityKind || (exports.PolicyActivityKind = {}));
// FIXME: Missing documentation for PolicyStatus
var PolicyStatus;
(function (PolicyStatus) {
    // FIXME: Missing documentation for Enabled
    PolicyStatus["Enabled"] = "Enabled";
    // FIXME: Missing documentation for Disabled
    PolicyStatus["Disabled"] = "Disabled";
    // FIXME: Missing documentation for Archived
    PolicyStatus["Archived"] = "Archived";
})(PolicyStatus = exports.PolicyStatus || (exports.PolicyStatus = {}));
/**
 * enumm:
 * * NotifyAndAudit
 * * RequestApproval
 */
var PolicyControlKind;
(function (PolicyControlKind) {
    // FIXME: Missing documentation for NotifyAndAudit
    PolicyControlKind["NotifyAndAudit"] = "NotifyAndAudit";
    // FIXME: Missing documentation for RequestApproval
    PolicyControlKind["RequestApproval"] = "RequestApproval";
})(PolicyControlKind = exports.PolicyControlKind || (exports.PolicyControlKind = {}));
// FIXME: Missing documentation for PolicyRuleKind
var PolicyRuleKind;
(function (PolicyRuleKind) {
    // FIXME: Missing documentation for AlwaysActivated
    PolicyRuleKind["AlwaysActivated"] = "AlwaysActivated";
    // FIXME: Missing documentation for PaymentAmountLimit
    PolicyRuleKind["PaymentAmountLimit"] = "PaymentAmountLimit";
    // FIXME: Missing documentation for TransferAmountLimit
    PolicyRuleKind["TransferAmountLimit"] = "TransferAmountLimit";
})(PolicyRuleKind = exports.PolicyRuleKind || (exports.PolicyRuleKind = {}));
// FIXME: Missing documentation for PolicyControlStatus
var PolicyControlStatus;
(function (PolicyControlStatus) {
    // FIXME: Missing documentation for Enabled
    PolicyControlStatus["Enabled"] = "Enabled";
    // FIXME: Missing documentation for Archived
    PolicyControlStatus["Archived"] = "Archived";
})(PolicyControlStatus = exports.PolicyControlStatus || (exports.PolicyControlStatus = {}));
// FIXME: Missing documentation for PolicyRuleStatus
var PolicyRuleStatus;
(function (PolicyRuleStatus) {
    // FIXME: Missing documentation for Enabled
    PolicyRuleStatus["Enabled"] = "Enabled";
    // FIXME: Missing documentation for Archived
    PolicyRuleStatus["Archived"] = "Archived";
})(PolicyRuleStatus = exports.PolicyRuleStatus || (exports.PolicyRuleStatus = {}));
// FIXME: Missing documentation for PolicyObjectFilterKind
var PolicyObjectFilterKind;
(function (PolicyObjectFilterKind) {
    // FIXME: Missing documentation for AssetAccount
    PolicyObjectFilterKind["AssetAccount"] = "AssetAccount";
    // FIXME: Missing documentation for PublicKey
    PolicyObjectFilterKind["PublicKey"] = "PublicKey";
    // FIXME: Missing documentation for Wallet
    PolicyObjectFilterKind["Wallet"] = "Wallet";
})(PolicyObjectFilterKind = exports.PolicyObjectFilterKind || (exports.PolicyObjectFilterKind = {}));
