// @dfns/sdk/codegen/datamodel/Auth/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthIdentityKind = exports.UserAuthKind = exports.AccessTokenKind = exports.ServerKind = exports.CredentialFactor = exports.FidoCredentialsTransportKind = exports.ApplicationKind = exports.AuthenticatorAttestationOptions = exports.AuthenticatorRequirementOptions = exports.UserKind = exports.CredentialKind = void 0;
// FIXME: Missing documentation for CredentialKind
var CredentialKind;
(function (CredentialKind) {
    // FIXME: Missing documentation for Fido2
    CredentialKind["Fido2"] = "Fido2";
    // FIXME: Missing documentation for Key
    CredentialKind["Key"] = "Key";
    // FIXME: Missing documentation for Password
    CredentialKind["Password"] = "Password";
    // FIXME: Missing documentation for Totp
    CredentialKind["Totp"] = "Totp";
    // FIXME: Missing documentation for RecoveryKey
    CredentialKind["RecoveryKey"] = "RecoveryKey";
})(CredentialKind = exports.CredentialKind || (exports.CredentialKind = {}));
// FIXME: Missing documentation for UserKind
var UserKind;
(function (UserKind) {
    // FIXME: Missing documentation for CustomerEmployee
    UserKind["CustomerEmployee"] = "CustomerEmployee";
    // FIXME: Missing documentation for DfnsStaff
    UserKind["DfnsStaff"] = "DfnsStaff";
    // FIXME: Missing documentation for EndUser
    UserKind["EndUser"] = "EndUser";
    // FIXME: Missing documentation for Pat
    UserKind["Pat"] = "Pat";
    // FIXME: Missing documentation for Application
    UserKind["Application"] = "Application";
    // FIXME: Missing documentation for ServiceAccount
    UserKind["ServiceAccount"] = "ServiceAccount";
})(UserKind = exports.UserKind || (exports.UserKind = {}));
// FIXME: Missing documentation for AuthenticatorRequirementOptions
var AuthenticatorRequirementOptions;
(function (AuthenticatorRequirementOptions) {
    // FIXME: Missing documentation for required
    AuthenticatorRequirementOptions["required"] = "required";
    // FIXME: Missing documentation for preferred
    AuthenticatorRequirementOptions["preferred"] = "preferred";
    // FIXME: Missing documentation for discouraged
    AuthenticatorRequirementOptions["discouraged"] = "discouraged";
})(AuthenticatorRequirementOptions = exports.AuthenticatorRequirementOptions || (exports.AuthenticatorRequirementOptions = {}));
// FIXME: Missing documentation for AuthenticatorAttestationOptions
var AuthenticatorAttestationOptions;
(function (AuthenticatorAttestationOptions) {
    // FIXME: Missing documentation for none
    AuthenticatorAttestationOptions["none"] = "none";
    // FIXME: Missing documentation for indirect
    AuthenticatorAttestationOptions["indirect"] = "indirect";
    // FIXME: Missing documentation for direct
    AuthenticatorAttestationOptions["direct"] = "direct";
    // FIXME: Missing documentation for enterprise
    AuthenticatorAttestationOptions["enterprise"] = "enterprise";
})(AuthenticatorAttestationOptions = exports.AuthenticatorAttestationOptions || (exports.AuthenticatorAttestationOptions = {}));
// FIXME: Missing documentation for ApplicationKind
var ApplicationKind;
(function (ApplicationKind) {
    // FIXME: Missing documentation for ServerSideApplication
    ApplicationKind["ServerSideApplication"] = "ServerSideApplication";
    // FIXME: Missing documentation for ClientSideApplication
    ApplicationKind["ClientSideApplication"] = "ClientSideApplication";
})(ApplicationKind = exports.ApplicationKind || (exports.ApplicationKind = {}));
// FIXME: Missing documentation for FidoCredentialsTransportKind
var FidoCredentialsTransportKind;
(function (FidoCredentialsTransportKind) {
    // FIXME: Missing documentation for usb
    FidoCredentialsTransportKind["usb"] = "usb";
    // FIXME: Missing documentation for nfc
    FidoCredentialsTransportKind["nfc"] = "nfc";
    // FIXME: Missing documentation for ble
    FidoCredentialsTransportKind["ble"] = "ble";
    // FIXME: Missing documentation for internal
    FidoCredentialsTransportKind["internal"] = "internal";
    // FIXME: Missing documentation for hybrid
    FidoCredentialsTransportKind["hybrid"] = "hybrid";
})(FidoCredentialsTransportKind = exports.FidoCredentialsTransportKind || (exports.FidoCredentialsTransportKind = {}));
// FIXME: Missing documentation for CredentialFactor
var CredentialFactor;
(function (CredentialFactor) {
    // FIXME: Missing documentation for first
    CredentialFactor["first"] = "first";
    // FIXME: Missing documentation for second
    CredentialFactor["second"] = "second";
    // FIXME: Missing documentation for either
    CredentialFactor["either"] = "either";
})(CredentialFactor = exports.CredentialFactor || (exports.CredentialFactor = {}));
// FIXME: Missing documentation for ServerKind
var ServerKind;
(function (ServerKind) {
    // FIXME: Missing documentation for Api
    ServerKind["Api"] = "Api";
    // FIXME: Missing documentation for Staff
    ServerKind["Staff"] = "Staff";
})(ServerKind = exports.ServerKind || (exports.ServerKind = {}));
// FIXME: Missing documentation for AccessTokenKind
var AccessTokenKind;
(function (AccessTokenKind) {
    // FIXME: Missing documentation for ServiceAccount
    AccessTokenKind["ServiceAccount"] = "ServiceAccount";
    // FIXME: Missing documentation for Pat
    AccessTokenKind["Pat"] = "Pat";
    // FIXME: Missing documentation for Application
    AccessTokenKind["Application"] = "Application";
})(AccessTokenKind = exports.AccessTokenKind || (exports.AccessTokenKind = {}));
// FIXME: Missing documentation for UserAuthKind
var UserAuthKind;
(function (UserAuthKind) {
    // FIXME: Missing documentation for EndUser
    UserAuthKind["EndUser"] = "EndUser";
    // FIXME: Missing documentation for CustomerEmployee
    UserAuthKind["CustomerEmployee"] = "CustomerEmployee";
    // FIXME: Missing documentation for DfnsStaff
    UserAuthKind["DfnsStaff"] = "DfnsStaff";
})(UserAuthKind = exports.UserAuthKind || (exports.UserAuthKind = {}));
// FIXME: Missing documentation for AuthIdentityKind
var AuthIdentityKind;
(function (AuthIdentityKind) {
    // FIXME: Missing documentation for DfnsStaffIdentity
    AuthIdentityKind["DfnsStaffIdentity"] = "DfnsStaffIdentity";
    // FIXME: Missing documentation for OrgEmployeeIdentity
    AuthIdentityKind["OrgEmployeeIdentity"] = "OrgEmployeeIdentity";
    // FIXME: Missing documentation for OrgApiKeyIdentity
    AuthIdentityKind["OrgApiKeyIdentity"] = "OrgApiKeyIdentity";
    // FIXME: Missing documentation for DfnsService
    AuthIdentityKind["DfnsService"] = "DfnsService";
})(AuthIdentityKind = exports.AuthIdentityKind || (exports.AuthIdentityKind = {}));
