// @dfns/sdk/codegen/datamodel/PublicKeys/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionStatus = exports.TransactionTemplateKind = exports.SignatureStatus = exports.PublicKeyStatus = void 0;
// FIXME: Missing documentation for PublicKeyStatus
var PublicKeyStatus;
(function (PublicKeyStatus) {
    // FIXME: Missing documentation for Active
    PublicKeyStatus["Active"] = "Active";
    // FIXME: Missing documentation for Disabled
    PublicKeyStatus["Disabled"] = "Disabled";
    // FIXME: Missing documentation for Compromised
    PublicKeyStatus["Compromised"] = "Compromised";
})(PublicKeyStatus = exports.PublicKeyStatus || (exports.PublicKeyStatus = {}));
// FIXME: Missing documentation for SignatureStatus
var SignatureStatus;
(function (SignatureStatus) {
    // FIXME: Missing documentation for Pending
    SignatureStatus["Pending"] = "Pending";
    // FIXME: Missing documentation for Executed
    SignatureStatus["Executed"] = "Executed";
    // FIXME: Missing documentation for Rejected
    SignatureStatus["Rejected"] = "Rejected";
    // FIXME: Missing documentation for Failed
    SignatureStatus["Failed"] = "Failed";
})(SignatureStatus = exports.SignatureStatus || (exports.SignatureStatus = {}));
// FIXME: Missing documentation for TransactionTemplateKind
var TransactionTemplateKind;
(function (TransactionTemplateKind) {
    // FIXME: Missing documentation for EvmGenericTx
    TransactionTemplateKind["EvmGenericTx"] = "EvmGenericTx";
    // FIXME: Missing documentation for CosmosIbcTransfer
    TransactionTemplateKind["CosmosIbcTransfer"] = "CosmosIbcTransfer";
    // FIXME: Missing documentation for SolanaTx
    TransactionTemplateKind["SolanaTx"] = "SolanaTx";
    // FIXME: Missing documentation for TronTx
    TransactionTemplateKind["TronTx"] = "TronTx";
    // FIXME: Missing documentation for TezosTx
    TransactionTemplateKind["TezosTx"] = "TezosTx";
})(TransactionTemplateKind = exports.TransactionTemplateKind || (exports.TransactionTemplateKind = {}));
// FIXME: Missing documentation for TransactionStatus
var TransactionStatus;
(function (TransactionStatus) {
    // FIXME: Missing documentation for Initiated
    TransactionStatus["Initiated"] = "Initiated";
    // FIXME: Missing documentation for PolicyAccepted
    TransactionStatus["PolicyAccepted"] = "PolicyAccepted";
    // FIXME: Missing documentation for PolicyRejected
    TransactionStatus["PolicyRejected"] = "PolicyRejected";
    // FIXME: Missing documentation for BroadcastPending
    TransactionStatus["BroadcastPending"] = "BroadcastPending";
    // FIXME: Missing documentation for BroadcastRejected
    TransactionStatus["BroadcastRejected"] = "BroadcastRejected";
    // FIXME: Missing documentation for Broadcasted
    TransactionStatus["Broadcasted"] = "Broadcasted";
    // FIXME: Missing documentation for Confirmed
    TransactionStatus["Confirmed"] = "Confirmed";
    // FIXME: Missing documentation for Failed
    TransactionStatus["Failed"] = "Failed";
})(TransactionStatus = exports.TransactionStatus || (exports.TransactionStatus = {}));
