// @dfns/sdk/codegen/datamodel/Assets/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentDirection = exports.AssetAccountPermissions = exports.AssetAccountAuthorizationKind = exports.PaymentStatus = exports.AssetAccountStatus = exports.PaymentInitiationStatus = exports.PaymentInstrumentKind = void 0;
// FIXME: Missing documentation for PaymentInstrumentKind
var PaymentInstrumentKind;
(function (PaymentInstrumentKind) {
    // FIXME: Missing documentation for DfnsAssetAccount
    PaymentInstrumentKind["DfnsAssetAccount"] = "DfnsAssetAccount";
    // FIXME: Missing documentation for BlockchainWalletAddress
    PaymentInstrumentKind["BlockchainWalletAddress"] = "BlockchainWalletAddress";
})(PaymentInstrumentKind = exports.PaymentInstrumentKind || (exports.PaymentInstrumentKind = {}));
// FIXME: Missing documentation for PaymentInitiationStatus
var PaymentInitiationStatus;
(function (PaymentInitiationStatus) {
    // FIXME: Missing documentation for Initiated
    PaymentInitiationStatus["Initiated"] = "Initiated";
    // FIXME: Missing documentation for Canceled
    PaymentInitiationStatus["Canceled"] = "Canceled";
    // FIXME: Missing documentation for Rejected
    PaymentInitiationStatus["Rejected"] = "Rejected";
})(PaymentInitiationStatus = exports.PaymentInitiationStatus || (exports.PaymentInitiationStatus = {}));
// FIXME: Missing documentation for AssetAccountStatus
var AssetAccountStatus;
(function (AssetAccountStatus) {
    //Initial state of `AssetAccount` entity, indicating that it’s being created at the moment.
    AssetAccountStatus["Creating"] = "Creating";
    // FIXME: Missing documentation for Enabled
    AssetAccountStatus["Enabled"] = "Enabled";
    // FIXME: Missing documentation for Failed
    AssetAccountStatus["Failed"] = "Failed";
})(AssetAccountStatus = exports.AssetAccountStatus || (exports.AssetAccountStatus = {}));
// FIXME: Missing documentation for PaymentStatus
var PaymentStatus;
(function (PaymentStatus) {
    //Payment order is received and initiated. It will execute later.
    PaymentStatus["Initiated"] = "Initiated";
    //Some Policy blocked the payment, it is not approved.
    PaymentStatus["Rejected"] = "Rejected";
    //Payment passed Policies, it is approved.
    PaymentStatus["Approved"] = "Approved";
    //Payment is successfully Executed and Broadcasted on the blockchain.
    PaymentStatus["Executed"] = "Executed";
    //Payment tried to execute, but failed doing so.
    PaymentStatus["Failed"] = "Failed";
    //Payment transaction is confirmed on the blockchain.
    PaymentStatus["Confirmed"] = "Confirmed";
})(PaymentStatus = exports.PaymentStatus || (exports.PaymentStatus = {}));
// FIXME: Missing documentation for AssetAccountAuthorizationKind
var AssetAccountAuthorizationKind;
(function (AssetAccountAuthorizationKind) {
    // FIXME: Missing documentation for Employee
    AssetAccountAuthorizationKind["Employee"] = "Employee";
    // FIXME: Missing documentation for Group
    AssetAccountAuthorizationKind["Group"] = "Group";
    // FIXME: Missing documentation for ApiKey
    AssetAccountAuthorizationKind["ApiKey"] = "ApiKey";
})(AssetAccountAuthorizationKind = exports.AssetAccountAuthorizationKind || (exports.AssetAccountAuthorizationKind = {}));
// FIXME: Missing documentation for AssetAccountPermissions
var AssetAccountPermissions;
(function (AssetAccountPermissions) {
    // FIXME: Missing documentation for InitiatePayments
    AssetAccountPermissions["InitiatePayments"] = "InitiatePayments";
    // FIXME: Missing documentation for ReadBalance
    AssetAccountPermissions["ReadBalance"] = "ReadBalance";
    // FIXME: Missing documentation for ReadPublicKey
    AssetAccountPermissions["ReadPublicKey"] = "ReadPublicKey";
})(AssetAccountPermissions = exports.AssetAccountPermissions || (exports.AssetAccountPermissions = {}));
// FIXME: Missing documentation for PaymentDirection
var PaymentDirection;
(function (PaymentDirection) {
    // FIXME: Missing documentation for Incoming
    PaymentDirection["Incoming"] = "Incoming";
    // FIXME: Missing documentation for Outgoing
    PaymentDirection["Outgoing"] = "Outgoing";
})(PaymentDirection = exports.PaymentDirection || (exports.PaymentDirection = {}));
