// @dfns/sdk/codegen/PolicyExecution/DelegatedPolicyExecutionClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DelegatedPolicyExecutionClient = void 0;
const baseAuthApi_1 = require("../../baseAuthApi");
const fetch_1 = require("../../utils/fetch");
const url_1 = require("../../utils/url");
class DelegatedPolicyExecutionClient {
    constructor(apiOptions) {
        this.apiOptions = apiOptions;
    }
    async listPolicyControlExecutions(request) {
        var _a;
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-control-executions', {
            path: {},
            query: (_a = request.query) !== null && _a !== void 0 ? _a : {},
        });
        const response = await (0, fetch_1.simpleFetch)(path, {
            method: 'GET',
            apiOptions: this.apiOptions,
        });
        return response.json();
    }
    async getPolicyControlExecutionById(request) {
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-control-executions/:policyControlExecutionId', {
            path: { policyControlExecutionId: request.policyControlExecutionId },
            query: {},
        });
        const response = await (0, fetch_1.simpleFetch)(path, {
            method: 'GET',
            apiOptions: this.apiOptions,
        });
        return response.json();
    }
    async updatePolicyControlExecutionInit(request) {
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-control-executions/:policyControlExecutionId', {
            path: { policyControlExecutionId: request.policyControlExecutionId },
            query: {},
        });
        const challenge = await baseAuthApi_1.BaseAuthApi.createUserActionChallenge({
            userActionHttpMethod: 'PUT',
            userActionHttpPath: path,
            userActionPayload: JSON.stringify(request.body),
            userActionServerKind: 'Api',
        }, this.apiOptions);
        return challenge;
    }
    async updatePolicyControlExecutionComplete(request, signedChallenge) {
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-control-executions/:policyControlExecutionId', {
            path: { policyControlExecutionId: request.policyControlExecutionId },
            query: {},
        });
        const { userAction } = await baseAuthApi_1.BaseAuthApi.signUserActionChallenge(signedChallenge, this.apiOptions);
        const response = await (0, fetch_1.simpleFetch)(path, {
            method: 'PUT',
            body: request.body,
            headers: { 'x-dfns-useraction': userAction },
            apiOptions: this.apiOptions,
        });
        return response.json();
    }
}
exports.DelegatedPolicyExecutionClient = DelegatedPolicyExecutionClient;
