// @dfns/sdk/codegen/datamodel/Wallets/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BlockchainNetwork = exports.SignatureStatus = exports.TransactionStatus = exports.TransferStatus = exports.TransferDirection = exports.EventKind = exports.WalletStatus = exports.SignatureKind = exports.TransactionKind = exports.TransferKind = void 0;
// FIXME: Missing documentation for TransferKind
var TransferKind;
(function (TransferKind) {
    // FIXME: Missing documentation for Native
    TransferKind["Native"] = "Native";
    // FIXME: Missing documentation for Erc20
    TransferKind["Erc20"] = "Erc20";
    // FIXME: Missing documentation for Erc721
    TransferKind["Erc721"] = "Erc721";
})(TransferKind = exports.TransferKind || (exports.TransferKind = {}));
// FIXME: Missing documentation for TransactionKind
var TransactionKind;
(function (TransactionKind) {
    // FIXME: Missing documentation for Evm
    TransactionKind["Evm"] = "Evm";
    // FIXME: Missing documentation for Eip1559
    TransactionKind["Eip1559"] = "Eip1559";
    // FIXME: Missing documentation for EvmLegacy
    TransactionKind["EvmLegacy"] = "EvmLegacy";
})(TransactionKind = exports.TransactionKind || (exports.TransactionKind = {}));
// FIXME: Missing documentation for SignatureKind
var SignatureKind;
(function (SignatureKind) {
    // FIXME: Missing documentation for Hash
    SignatureKind["Hash"] = "Hash";
    // FIXME: Missing documentation for Eip712
    SignatureKind["Eip712"] = "Eip712";
})(SignatureKind = exports.SignatureKind || (exports.SignatureKind = {}));
// FIXME: Missing documentation for WalletStatus
var WalletStatus;
(function (WalletStatus) {
    // FIXME: Missing documentation for Active
    WalletStatus["Active"] = "Active";
    //Initial state of `AssetAccount` entity, indicating that it’s being created at the moment.
    WalletStatus["Creating"] = "Creating";
    // FIXME: Missing documentation for Failed
    WalletStatus["Failed"] = "Failed";
})(WalletStatus = exports.WalletStatus || (exports.WalletStatus = {}));
// FIXME: Missing documentation for EventKind
var EventKind;
(function (EventKind) {
    // FIXME: Missing documentation for NativeTransfer
    EventKind["NativeTransfer"] = "NativeTransfer";
    // FIXME: Missing documentation for Erc20Transfer
    EventKind["Erc20Transfer"] = "Erc20Transfer";
    // FIXME: Missing documentation for Erc721Transfer
    EventKind["Erc721Transfer"] = "Erc721Transfer";
})(EventKind = exports.EventKind || (exports.EventKind = {}));
// FIXME: Missing documentation for TransferDirection
var TransferDirection;
(function (TransferDirection) {
    // FIXME: Missing documentation for In
    TransferDirection["In"] = "In";
    // FIXME: Missing documentation for Out
    TransferDirection["Out"] = "Out";
})(TransferDirection = exports.TransferDirection || (exports.TransferDirection = {}));
// FIXME: Missing documentation for TransferStatus
var TransferStatus;
(function (TransferStatus) {
    // FIXME: Missing documentation for Pending
    TransferStatus["Pending"] = "Pending";
    // FIXME: Missing documentation for Broadcasted
    TransferStatus["Broadcasted"] = "Broadcasted";
    // FIXME: Missing documentation for Confirmed
    TransferStatus["Confirmed"] = "Confirmed";
    // FIXME: Missing documentation for Failed
    TransferStatus["Failed"] = "Failed";
})(TransferStatus = exports.TransferStatus || (exports.TransferStatus = {}));
// FIXME: Missing documentation for TransactionStatus
var TransactionStatus;
(function (TransactionStatus) {
    // FIXME: Missing documentation for Pending
    TransactionStatus["Pending"] = "Pending";
    // FIXME: Missing documentation for Broadcasted
    TransactionStatus["Broadcasted"] = "Broadcasted";
    // FIXME: Missing documentation for Confirmed
    TransactionStatus["Confirmed"] = "Confirmed";
    // FIXME: Missing documentation for Failed
    TransactionStatus["Failed"] = "Failed";
})(TransactionStatus = exports.TransactionStatus || (exports.TransactionStatus = {}));
// FIXME: Missing documentation for SignatureStatus
var SignatureStatus;
(function (SignatureStatus) {
    // FIXME: Missing documentation for Pending
    SignatureStatus["Pending"] = "Pending";
    // FIXME: Missing documentation for Signed
    SignatureStatus["Signed"] = "Signed";
    // FIXME: Missing documentation for Confirmed
    SignatureStatus["Confirmed"] = "Confirmed";
    // FIXME: Missing documentation for Failed
    SignatureStatus["Failed"] = "Failed";
})(SignatureStatus = exports.SignatureStatus || (exports.SignatureStatus = {}));
// FIXME: Missing documentation for BlockchainNetwork
var BlockchainNetwork;
(function (BlockchainNetwork) {
    // FIXME: Missing documentation for Bitcoin
    BlockchainNetwork["Bitcoin"] = "Bitcoin";
    // FIXME: Missing documentation for BitcoinTestnet
    BlockchainNetwork["BitcoinTestnet"] = "BitcoinTestnet";
    // FIXME: Missing documentation for Bsc
    BlockchainNetwork["Bsc"] = "Bsc";
    // FIXME: Missing documentation for BscTestnet
    BlockchainNetwork["BscTestnet"] = "BscTestnet";
    // FIXME: Missing documentation for Ethereum
    BlockchainNetwork["Ethereum"] = "Ethereum";
    // FIXME: Missing documentation for EthereumGoerli
    BlockchainNetwork["EthereumGoerli"] = "EthereumGoerli";
    // FIXME: Missing documentation for EthereumSepolia
    BlockchainNetwork["EthereumSepolia"] = "EthereumSepolia";
    // FIXME: Missing documentation for Polygon
    BlockchainNetwork["Polygon"] = "Polygon";
    // FIXME: Missing documentation for PolygonMumbai
    BlockchainNetwork["PolygonMumbai"] = "PolygonMumbai";
    // FIXME: Missing documentation for Ripple
    BlockchainNetwork["Ripple"] = "Ripple";
    // FIXME: Missing documentation for RippleTestnet
    BlockchainNetwork["RippleTestnet"] = "RippleTestnet";
    // FIXME: Missing documentation for Solana
    BlockchainNetwork["Solana"] = "Solana";
    // FIXME: Missing documentation for SolanaDevnet
    BlockchainNetwork["SolanaDevnet"] = "SolanaDevnet";
    // FIXME: Missing documentation for Tron
    BlockchainNetwork["Tron"] = "Tron";
    // FIXME: Missing documentation for TronShasta
    BlockchainNetwork["TronShasta"] = "TronShasta";
})(BlockchainNetwork = exports.BlockchainNetwork || (exports.BlockchainNetwork = {}));
