// @dfns/sdk/codegen/datamodel/Callbacks/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallbackSubscriptionStatus = exports.CallbackEventStatus = exports.CallbackEventKind = void 0;
// FIXME: Missing documentation for CallbackEventKind
var CallbackEventKind;
(function (CallbackEventKind) {
    // FIXME: Missing documentation for PaymentInitiated
    CallbackEventKind["PaymentInitiated"] = "PaymentInitiated";
    // FIXME: Missing documentation for PaymentExecuted
    CallbackEventKind["PaymentExecuted"] = "PaymentExecuted";
    // FIXME: Missing documentation for PaymentConfirmed
    CallbackEventKind["PaymentConfirmed"] = "PaymentConfirmed";
    // FIXME: Missing documentation for PaymentReceived
    CallbackEventKind["PaymentReceived"] = "PaymentReceived";
    // FIXME: Missing documentation for TransactionBroadcasted
    CallbackEventKind["TransactionBroadcasted"] = "TransactionBroadcasted";
    // FIXME: Missing documentation for WalletCreated
    CallbackEventKind["WalletCreated"] = "WalletCreated";
    // FIXME: Missing documentation for PolicyActivated
    CallbackEventKind["PolicyActivated"] = "PolicyActivated";
})(CallbackEventKind = exports.CallbackEventKind || (exports.CallbackEventKind = {}));
// FIXME: Missing documentation for CallbackEventStatus
var CallbackEventStatus;
(function (CallbackEventStatus) {
    // FIXME: Missing documentation for Sent
    CallbackEventStatus["Sent"] = "Sent";
    // FIXME: Missing documentation for Failed
    CallbackEventStatus["Failed"] = "Failed";
})(CallbackEventStatus = exports.CallbackEventStatus || (exports.CallbackEventStatus = {}));
// FIXME: Missing documentation for CallbackSubscriptionStatus
var CallbackSubscriptionStatus;
(function (CallbackSubscriptionStatus) {
    // FIXME: Missing documentation for Enabled
    CallbackSubscriptionStatus["Enabled"] = "Enabled";
    // FIXME: Missing documentation for Disabled
    CallbackSubscriptionStatus["Disabled"] = "Disabled";
    // FIXME: Missing documentation for Archived
    CallbackSubscriptionStatus["Archived"] = "Archived";
})(CallbackSubscriptionStatus = exports.CallbackSubscriptionStatus || (exports.CallbackSubscriptionStatus = {}));
