// @dfns/sdk/codegen/datamodel/Blockchains/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeeEstimationKind = exports.SolInstructionKind = exports.XrpTransactionKind = exports.TrxTransactionKind = exports.TransactionStatus = exports.EvmLogKind = void 0;
// FIXME: Missing documentation for EvmLogKind
var EvmLogKind;
(function (EvmLogKind) {
    // FIXME: Missing documentation for Unknown
    EvmLogKind["Unknown"] = "Unknown";
    // FIXME: Missing documentation for Erc20
    EvmLogKind["Erc20"] = "Erc20";
    // FIXME: Missing documentation for Erc721
    EvmLogKind["Erc721"] = "Erc721";
})(EvmLogKind = exports.EvmLogKind || (exports.EvmLogKind = {}));
// FIXME: Missing documentation for TransactionStatus
var TransactionStatus;
(function (TransactionStatus) {
    // FIXME: Missing documentation for SUCCESS
    TransactionStatus["SUCCESS"] = "SUCCESS";
    // FIXME: Missing documentation for FAILED
    TransactionStatus["FAILED"] = "FAILED";
})(TransactionStatus = exports.TransactionStatus || (exports.TransactionStatus = {}));
// FIXME: Missing documentation for TrxTransactionKind
var TrxTransactionKind;
(function (TrxTransactionKind) {
    // FIXME: Missing documentation for TransferContract
    TrxTransactionKind["TransferContract"] = "TransferContract";
    // FIXME: Missing documentation for TransferAssetContract
    TrxTransactionKind["TransferAssetContract"] = "TransferAssetContract";
    // FIXME: Missing documentation for TriggerSmartContract
    TrxTransactionKind["TriggerSmartContract"] = "TriggerSmartContract";
})(TrxTransactionKind = exports.TrxTransactionKind || (exports.TrxTransactionKind = {}));
// FIXME: Missing documentation for XrpTransactionKind
var XrpTransactionKind;
(function (XrpTransactionKind) {
    // FIXME: Missing documentation for Payment
    XrpTransactionKind["Payment"] = "Payment";
})(XrpTransactionKind = exports.XrpTransactionKind || (exports.XrpTransactionKind = {}));
// FIXME: Missing documentation for SolInstructionKind
var SolInstructionKind;
(function (SolInstructionKind) {
    // FIXME: Missing documentation for SystemTransfer
    SolInstructionKind["SystemTransfer"] = "SystemTransfer";
    // FIXME: Missing documentation for TokenTransfer
    SolInstructionKind["TokenTransfer"] = "TokenTransfer";
})(SolInstructionKind = exports.SolInstructionKind || (exports.SolInstructionKind = {}));
// FIXME: Missing documentation for FeeEstimationKind
var FeeEstimationKind;
(function (FeeEstimationKind) {
    // FIXME: Missing documentation for Eip1559
    FeeEstimationKind["Eip1559"] = "Eip1559";
    // FIXME: Missing documentation for EthLegacy
    FeeEstimationKind["EthLegacy"] = "EthLegacy";
})(FeeEstimationKind = exports.FeeEstimationKind || (exports.FeeEstimationKind = {}));
