// @dfns/sdk/codegen/PolicyManagement/DelegatedPolicyManagementClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DelegatedPolicyManagementClient = void 0;
const baseAuthApi_1 = require("../../baseAuthApi");
const fetch_1 = require("../../utils/fetch");
const url_1 = require("../../utils/url");
class DelegatedPolicyManagementClient {
    constructor(apiOptions) {
        this.apiOptions = apiOptions;
    }
    async createPolicyInit(request) {
        const path = (0, url_1.buildPathAndQuery)('/policies', {
            path: {},
            query: {},
        });
        const challenge = await baseAuthApi_1.BaseAuthApi.createUserActionChallenge({
            userActionHttpMethod: 'POST',
            userActionHttpPath: path,
            userActionPayload: JSON.stringify(request.body),
            userActionServerKind: 'Api',
        }, this.apiOptions);
        return challenge;
    }
    async createPolicyComplete(request, signedChallenge) {
        const path = (0, url_1.buildPathAndQuery)('/policies', {
            path: {},
            query: {},
        });
        const { userAction } = await baseAuthApi_1.BaseAuthApi.signUserActionChallenge(signedChallenge, this.apiOptions);
        const response = await (0, fetch_1.simpleFetch)(path, {
            method: 'POST',
            body: request.body,
            headers: { 'x-dfns-useraction': userAction },
            apiOptions: this.apiOptions,
        });
        return response.json();
    }
    async getPolicyById(request) {
        const path = (0, url_1.buildPathAndQuery)('/policies/:policyId', {
            path: { policyId: request.policyId },
            query: {},
        });
        const response = await (0, fetch_1.simpleFetch)(path, {
            method: 'GET',
            apiOptions: this.apiOptions,
        });
        return response.json();
    }
    async listPolicies() {
        const path = (0, url_1.buildPathAndQuery)('/policies', {
            path: {},
            query: {},
        });
        const response = await (0, fetch_1.simpleFetch)(path, {
            method: 'GET',
            apiOptions: this.apiOptions,
        });
        return response.json();
    }
    async archivePolicyInit(request) {
        const path = (0, url_1.buildPathAndQuery)('/policies/:policyId', {
            path: { policyId: request.policyId },
            query: {},
        });
        const challenge = await baseAuthApi_1.BaseAuthApi.createUserActionChallenge({
            userActionHttpMethod: 'DELETE',
            userActionHttpPath: path,
            userActionPayload: JSON.stringify({}),
            userActionServerKind: 'Api',
        }, this.apiOptions);
        return challenge;
    }
    async archivePolicyComplete(request, signedChallenge) {
        const path = (0, url_1.buildPathAndQuery)('/policies/:policyId', {
            path: { policyId: request.policyId },
            query: {},
        });
        const { userAction } = await baseAuthApi_1.BaseAuthApi.signUserActionChallenge(signedChallenge, this.apiOptions);
        const response = await (0, fetch_1.simpleFetch)(path, {
            method: 'DELETE',
            body: {},
            headers: { 'x-dfns-useraction': userAction },
            apiOptions: this.apiOptions,
        });
        return response.json();
    }
    async createPolicyControlInit(request) {
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-controls', {
            path: {},
            query: {},
        });
        const challenge = await baseAuthApi_1.BaseAuthApi.createUserActionChallenge({
            userActionHttpMethod: 'POST',
            userActionHttpPath: path,
            userActionPayload: JSON.stringify(request.body),
            userActionServerKind: 'Api',
        }, this.apiOptions);
        return challenge;
    }
    async createPolicyControlComplete(request, signedChallenge) {
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-controls', {
            path: {},
            query: {},
        });
        const { userAction } = await baseAuthApi_1.BaseAuthApi.signUserActionChallenge(signedChallenge, this.apiOptions);
        const response = await (0, fetch_1.simpleFetch)(path, {
            method: 'POST',
            body: request.body,
            headers: { 'x-dfns-useraction': userAction },
            apiOptions: this.apiOptions,
        });
        return response.json();
    }
    async getPolicyControlById(request) {
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-controls/:policyControlId', {
            path: { policyControlId: request.policyControlId },
            query: {},
        });
        const response = await (0, fetch_1.simpleFetch)(path, {
            method: 'GET',
            apiOptions: this.apiOptions,
        });
        return response.json();
    }
    async listPolicyControls() {
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-controls', {
            path: {},
            query: {},
        });
        const response = await (0, fetch_1.simpleFetch)(path, {
            method: 'GET',
            apiOptions: this.apiOptions,
        });
        return response.json();
    }
    async archivePolicyControlInit(request) {
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-controls/:policyControlId', {
            path: { policyControlId: request.policyControlId },
            query: {},
        });
        const challenge = await baseAuthApi_1.BaseAuthApi.createUserActionChallenge({
            userActionHttpMethod: 'DELETE',
            userActionHttpPath: path,
            userActionPayload: JSON.stringify({}),
            userActionServerKind: 'Api',
        }, this.apiOptions);
        return challenge;
    }
    async archivePolicyControlComplete(request, signedChallenge) {
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-controls/:policyControlId', {
            path: { policyControlId: request.policyControlId },
            query: {},
        });
        const { userAction } = await baseAuthApi_1.BaseAuthApi.signUserActionChallenge(signedChallenge, this.apiOptions);
        const response = await (0, fetch_1.simpleFetch)(path, {
            method: 'DELETE',
            body: {},
            headers: { 'x-dfns-useraction': userAction },
            apiOptions: this.apiOptions,
        });
        return response.json();
    }
    async createPolicyRuleInit(request) {
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-rules', {
            path: {},
            query: {},
        });
        const challenge = await baseAuthApi_1.BaseAuthApi.createUserActionChallenge({
            userActionHttpMethod: 'POST',
            userActionHttpPath: path,
            userActionPayload: JSON.stringify(request.body),
            userActionServerKind: 'Api',
        }, this.apiOptions);
        return challenge;
    }
    async createPolicyRuleComplete(request, signedChallenge) {
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-rules', {
            path: {},
            query: {},
        });
        const { userAction } = await baseAuthApi_1.BaseAuthApi.signUserActionChallenge(signedChallenge, this.apiOptions);
        const response = await (0, fetch_1.simpleFetch)(path, {
            method: 'POST',
            body: request.body,
            headers: { 'x-dfns-useraction': userAction },
            apiOptions: this.apiOptions,
        });
        return response.json();
    }
    async getPolicyRuleById(request) {
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-rules/:policyRuleId', {
            path: { policyRuleId: request.policyRuleId },
            query: {},
        });
        const response = await (0, fetch_1.simpleFetch)(path, {
            method: 'GET',
            apiOptions: this.apiOptions,
        });
        return response.json();
    }
    async listPolicyRules(request) {
        var _a;
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-rules', {
            path: {},
            query: (_a = request.query) !== null && _a !== void 0 ? _a : {},
        });
        const response = await (0, fetch_1.simpleFetch)(path, {
            method: 'GET',
            apiOptions: this.apiOptions,
        });
        return response.json();
    }
    async archivePolicyRuleInit(request) {
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-rules/:policyRuleId', {
            path: { policyRuleId: request.policyRuleId },
            query: {},
        });
        const challenge = await baseAuthApi_1.BaseAuthApi.createUserActionChallenge({
            userActionHttpMethod: 'DELETE',
            userActionHttpPath: path,
            userActionPayload: JSON.stringify({}),
            userActionServerKind: 'Api',
        }, this.apiOptions);
        return challenge;
    }
    async archivePolicyRuleComplete(request, signedChallenge) {
        const path = (0, url_1.buildPathAndQuery)('/policies/policy-rules/:policyRuleId', {
            path: { policyRuleId: request.policyRuleId },
            query: {},
        });
        const { userAction } = await baseAuthApi_1.BaseAuthApi.signUserActionChallenge(signedChallenge, this.apiOptions);
        const response = await (0, fetch_1.simpleFetch)(path, {
            method: 'DELETE',
            body: {},
            headers: { 'x-dfns-useraction': userAction },
            apiOptions: this.apiOptions,
        });
        return response.json();
    }
}
exports.DelegatedPolicyManagementClient = DelegatedPolicyManagementClient;
