// @dfns/sdk/codegen/datamodel/Permissions/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceKind = exports.OwnerKind = exports.PermissionStatus = exports.FieldEvaluatorKind = void 0;
// FIXME: Missing documentation for FieldEvaluatorKind
var FieldEvaluatorKind;
(function (FieldEvaluatorKind) {
    // FIXME: Missing documentation for ContainsAll
    FieldEvaluatorKind["ContainsAll"] = "ContainsAll";
    // FIXME: Missing documentation for ContainsOneOf
    FieldEvaluatorKind["ContainsOneOf"] = "ContainsOneOf";
    // FIXME: Missing documentation for DateAfter
    FieldEvaluatorKind["DateAfter"] = "DateAfter";
    // FIXME: Missing documentation for DateBefore
    FieldEvaluatorKind["DateBefore"] = "DateBefore";
    // FIXME: Missing documentation for Is
    FieldEvaluatorKind["Is"] = "Is";
    // FIXME: Missing documentation for ValueGreaterThan
    FieldEvaluatorKind["ValueGreaterThan"] = "ValueGreaterThan";
    // FIXME: Missing documentation for ValueLessThan
    FieldEvaluatorKind["ValueLessThan"] = "ValueLessThan";
})(FieldEvaluatorKind = exports.FieldEvaluatorKind || (exports.FieldEvaluatorKind = {}));
// FIXME: Missing documentation for PermissionStatus
var PermissionStatus;
(function (PermissionStatus) {
    // FIXME: Missing documentation for Active
    PermissionStatus["Active"] = "Active";
})(PermissionStatus = exports.PermissionStatus || (exports.PermissionStatus = {}));
// FIXME: Missing documentation for OwnerKind
var OwnerKind;
(function (OwnerKind) {
    // FIXME: Missing documentation for LegalPerson
    OwnerKind["LegalPerson"] = "LegalPerson";
    // FIXME: Missing documentation for NaturalPerson
    OwnerKind["NaturalPerson"] = "NaturalPerson";
})(OwnerKind = exports.OwnerKind || (exports.OwnerKind = {}));
// FIXME: Missing documentation for ResourceKind
var ResourceKind;
(function (ResourceKind) {
    // FIXME: Missing documentation for Wallet
    ResourceKind["Wallet"] = "Wallet";
    // FIXME: Missing documentation for PublicKey
    ResourceKind["PublicKey"] = "PublicKey";
})(ResourceKind = exports.ResourceKind || (exports.ResourceKind = {}));
