// @gelatonetwork/relay-sdk/dist/lib/erc2771/utils/mapRequestToStruct.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.mapRequestToStruct = void 0;
const ethers_1 = require("ethers");
const utils_1 = require("ethers/lib/utils");
const mapRequestToStruct = (request, override) => __awaiter(void 0, void 0, void 0, function* () {
    var _a, _b;
    if (!override.userNonce && !request.userNonce) {
        throw new Error(`userNonce is not found in the request, nor fetched`);
    }
    if (!override.userDeadline && !request.userDeadline) {
        throw new Error(`userDeadline is not found in the request, nor fetched`);
    }
    return {
        userNonce: 
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        (_a = override.userNonce) !== null && _a !== void 0 ? _a : ethers_1.BigNumber.from(request.userNonce).toString(),
        userDeadline: 
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        (_b = override.userDeadline) !== null && _b !== void 0 ? _b : ethers_1.BigNumber.from(request.userDeadline).toString(),
        chainId: ethers_1.BigNumber.from(request.chainId).toString(),
        target: (0, utils_1.getAddress)(request.target),
        data: request.data,
        user: (0, utils_1.getAddress)(request.user),
    };
});
exports.mapRequestToStruct = mapRequestToStruct;
